test/libevmasm/Assembler.cpp

```diff
@@ -254,6 +255,30 @@ BOOST_AUTO_TEST_CASE(immutable)
 	);
 }
 
+BOOST_AUTO_TEST_CASE(subobject_encode_decode)
+{
+	Assembly assembly;
+	auto root_asm = make_shared<CharStream>("lorem ipsum", "root.asm");
+
+	Assembly subAsm;
+	auto sub_asm = make_shared<CharStream>("lorem ipsum", "sub.asm");
+	shared_ptr<Assembly> _subAsmPtr = make_shared<Assembly>(subAsm);
+
+	Assembly subSubAsm;
+	auto sub_sub_asm = make_shared<CharStream>("lorem ipsum", "sub_sub.asm");
+	shared_ptr<Assembly> _subSubAsmPtr = make_shared<Assembly>(subSubAsm);
+
+	auto sub = assembly.appendSubroutine(_subAsmPtr);
+	auto subSub = subAsm.appendSubroutine(_subSubAsmPtr);
```

Comment:
This a case where I wouldn't use `auto`, because now I have to look up what types `sub` and `subSub` are :) `Assembly`, probably?

Reply:
AssemblyItem, will fix now as I agree with such auto rules like you just explained :)

Reply:
I also think that these variables are not needed at all, can be changed to only function call.